.travis.yml

```diff
@@ -1,14 +1,20 @@
 language: cpp
 
-os:
-  - linux
-  - osx
-
-env:
-  - variant=test_release coverage=1 docs=1
-  - variant=test_debug sim=1
-  - variant=test_barebones
-  - analyze=1
+matrix:
+  include:
+    - env: variant=test_release coverage=1
+    - env: variant=test_debug sim=1
+    - env: variant=test_barebones
+    - env: analyze=1
+    - os: osx
+      osx_image: xcode6.4
+      env: variant=test_release docs=1
+    - os: osx
+      osx_image: xcode6.4
+      env: variant=test_debug
+    - os: osx
+      osx_image: xcode6.4
+      env: variant=test_barebones
```

Comment:
does the default osx image have a too old version of clang?
With this change, I imagine there are some conditions further down that can be cleaned up as well, for instance, currently coverage is disabled for OSX by additional comparisons, as well as analyze=1